### DISCUSSION TITLE
Moving cursor with keybindings behaves different than other macOS text editor

### DISCUSSION THREAD

**Discussion by Ehlster** [krzyzanowskim/STTextView#50] [Questions]
Moving cursors behaves different in STTextView than in NSTextView or other macOS text editors, e.g. TextEdit app or Xcode:

1. Move to end of line: When you move to the end of a line (e.g. via `ctrl` + `e`), then the cursor jumps to the end of the line as expected. Then, if you press it a second time, I expect that nothing should happen because the cursor is already at the end of the line. This is also the behaviour in NSTextView / other text editors. But in STTextView, the cursor jumps to the begining of the next line.
2. Move to beginning of line:When you move to the beginning of a line (e.g. via `ctrl` + `a`), then the cursor jumps to the beginning of the line as expected. Then, if you press it a second time, I expect that nothing should happen because the cursor is already at the beginning of the line. This is also the behaviour in NSTextView / other text editors. But in STTextView, the cursor jumps to the end of the previous line.

The following test cases which fail describe the behaviour: 
```swift
class NavigationTestCase: XCTestCase {
    func testMoveEndOfLine() {
        let textView = STTextView()
        textView.setAttributedString(.init("1\n2\n3"))
        textView.setSelectedRange(NSRange(location: 0, length: 0))

        textView.moveToEndOfLine(nil)
        XCTAssertEqual(NSRange(location: 1, length: 0), textView.selectedRange())
        // Already at end of line, so the cursor should stay.
        textView.moveToEndOfLine(nil)
        XCTAssertEqual(NSRange(location: 1, length: 0), textView.selectedRange())
    }

    func testMoveBeginnningOfLine() {
        let textView = STTextView()
        textView.setAttributedString(.init("1\n2\n3"))
        textView.setSelectedRange(NSRange(location: 3, length: 0))

        textView.moveToBeginningOfLine(nil)
        XCTAssertEqual(NSRange(location: 2, length: 0), textView.selectedRange())
        // Already at the beginning of line, so the cursor should stay.
        textView.moveToBeginningOfLine(nil)
        XCTAssertEqual(NSRange(location: 2, length: 0), textView.selectedRange())
    }
}

extension NSTextView {
    func setAttributedString(_ attributedString: NSAttributedString) {
        textStorage?.setAttributedString(attributedString)
    }
}
```

If you replace `STTextView()` with `NSTextView()`, then the tests will pass.

**Comment by krzyzanowskim** [comment]
makes sense.

if you want, I'd welcome this change (with these tests) if you have some spare time.

**Reply by Ehlster** [reply]
If I'm not mistaken, then this should be a rather straightforward change. I can contribute a fix in the upcoming days.
